### PR TITLE
Fixed and simplified array_replace description

### PR DESCRIPTION
**Pull request by cartok**
Relates to: https://github.com/php/doc-de/pull/76

### CODE REVIEW

File: reference/array/functions/array-replace.xml [hunk 1]
```diff
@@ -13,18 +13,13 @@
...
-   in order, the later arrays overwriting the previous values.
+   <function>array_replace</function> is not mutating
   </para>
   <para>
-   <function>array_replace</function> is not recursive : it will replace
-   values in the first array by whatever type is in the second array.
+   <function>array_replace</function> is not recursive
+  </para>
+  <para>
+   <function>array_replace</function> creates a new array by joining all given arrays in given order.
```
Comment (cmb69) [inline reference/array/functions/array-replace.xml:22]: Wouldn't that be the description of `array_merge()`? It is certainly not what `array_replace()` is doing; see e.g. https://3v4l.org/XfXpa.

It seems to me that the current description is a bit verbose, but basically correct; it is only misleading, that the description suggests that the first array would be mutated. Maybe similar wording as for `str_replace()` might be used.

Comment (cartok) [reply reference/array/functions/array-replace.xml:22]: Yea ok, "joining" is propably not the best fitting word here.  

What about this?
```
creates a new array by joining all given arrays in given order, replacing values of already existing keys.
```

In the end the arrays are joined but it goes by keys, not indices. They not are just concatenated like in array_merge

File: reference/array/functions/array-replace.xml [hunk 2]
```diff
@@ -13,18 +13,13 @@
...
-   in order, the later arrays overwriting the previous values.
+   <function>array_replace</function> is not mutating
   </para>
   <para>
-   <function>array_replace</function> is not recursive : it will replace
-   values in the first array by whatever type is in the second array.
+   <function>array_replace</function> is not recursive
+  </para>
+  <para>
+   <function>array_replace</function> creates a new array by joining all given arrays in given order, replacing values of already existing keys.
```
[the remark this answers is not in the document]
Comment (cartok) [reply reference/array/functions/array-replace.xml:22]: For me its implicit that the latter arrays value is used from the former short description. Another thing is that it does not only assigns keys to the new array

[the remark this answers is not in the document]
Comment (Crell) [reply reference/array/functions/array-replace.xml:22]: A little redundancy here is fine, especially as it's in a different section.  Good point on the key/value question. I've updated the suggestion comment above.

**Review by cmb69 — COMMENTED** [review]
This would be rendered as:

Screenshot 2022-12-26 145520

This doesn't look like any improvement to me.

**Comment by cartok** [comment]
> This would be rendered as:
> 
> Screenshot 2022-12-26 145520 [ Screenshot 2022-12-26 145520 ](https://user-images.githubusercontent.com/2306138/209556109-f9e2febf-4029-489b-b251-da4662ca36e1.gif) [ ](https://user-images.githubusercontent.com/2306138/209556109-f9e2febf-4029-489b-b251-da4662ca36e1.gif)
> 
> This doesn't look like any improvement to me.

Why not, formatting / display or content?

**Comment by cmb69** [comment]
> Why not, formatting / display or content?

Almost everything. If you describe something, it doesn't make much sense to start telling what it is not (let alone that a sentence should end with a period, and there rarely is the need to have single sentence paragraphs). Furthermore, "joining" might easily mislead readers to associate that with https://www.php.net/join (yeah, that function is still a thing).
In my opinon, [@Crell's suggestion](https://github.com/php/doc-en/pull/1470#discussion_r907809356) as well as [my suggestion](https://github.com/php/doc-en/pull/1470#discussion_r830216752) are better than what we have here.

**Comment by cartok** [comment]
> > Why not, formatting / display or content?
> 
> Almost everything. If you describe something, it doesn't make much sense to start telling what it is not (let alone that a sentence should end with a period, and there rarely is the need to have single sentence paragraphs). Furthermore, "joining" might easily mislead readers to associate that with https://www.php.net/join (yeah, that function is still a thing). In my opinon, [@Crell's suggestion](https://github.com/php/doc-en/pull/1470#discussion_r907809356) as well as [my suggestion](https://github.com/php/doc-en/pull/1470#discussion_r830216752) are better than what we have here.

If you think so. I rather want to read short facts and not complicated stories on such a simple thing. Feel free to close it.

**Pull request closed without merging**